### PR TITLE
add DeepES of PaddleLite

### PR DESCRIPTION
**Pull request by zenghsh3**
- DeepES of PaddleLite
- CartPole demo of PaddleLite

### CODE REVIEW

File: deepes/src/paddle/es_agent.cc [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  _noise = new float [_param_size];
+  _neg_gradients = new float [_param_size];
+}
+
+std::shared_ptr<ESAgent> ESAgent::clone() {
+  std::shared_ptr<PaddlePredictor> new_sample_predictor = _predictor->Clone();
+
+  std::shared_ptr<ESAgent> new_agent = std::make_shared<ESAgent>();
+
+  float* new_noise = new float [_param_size];
```
Comment (TomorrowIsAnOtherDay) [inline deepes/src/paddle/es_agent.cc]: rename `new_niose` to `noise`

File: deepes/include/paddle/es_agent.h [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+   * @brief Get paddle predict
+   *
+   * if _is_sampling_agent is true, will return predictor with added noise;
+   * if _is_sampling_agent is false, will return predictor without added noise.
+   */
+  std::shared_ptr<PaddlePredictor> get_predictor();
+
+ private:
+  std::shared_ptr<PaddlePredictor> _predictor;
+  std::shared_ptr<PaddlePredictor> _sample_predictor;
```
Comment (TomorrowIsAnOtherDay) [inline deepes/include/paddle/es_agent.h]: Please rename `_sample_predictor` to `_sampling_predictor`, as we are using the name `sampling_method`.

File: deepes/include/paddle/es_agent.h [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  std::shared_ptr<SamplingMethod> _sampling_method;
+  std::shared_ptr<Optimizer> _optimizer;
+  std::shared_ptr<DeepESConfig> _config;
+  int64_t _param_size;
+  std::vector<std::string> _param_names;
+  // malloc memory of noise and neg_gradients in advance.
+  float* _noise;
+  float* _neg_gradients;
+
+  int64_t _calculate_param_size();
```
Comment (TomorrowIsAnOtherDay) [inline deepes/include/paddle/es_agent.h]: Please move this function before the definitions of class variables.

File: deepes/src/paddle/es_agent.cc (hunk 1 above)
Comment (zenghsh3) [reply deepes/src/paddle/es_agent.cc]: done

File: deepes/include/paddle/es_agent.h (hunk 2 above)
Comment (zenghsh3) [reply deepes/include/paddle/es_agent.h]: done

File: deepes/include/paddle/es_agent.h (hunk 3 above)
Comment (zenghsh3) [reply deepes/include/paddle/es_agent.h]: done

**Review by TomorrowIsAnOtherDay — APPROVED** [review]
LGTM

**Pull request merged**